lib/java/ydoc-server/src/main/java/org/enso/ydoc/polyfill/web/Target_Executors.java

```diff
@@ -0,0 +1,36 @@
+package org.enso.ydoc.polyfill.web;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+@TargetClass(className = "io.helidon.webserver.ServerListener")
+final class Target_ServerListener {
+  @Substitute
+  private static ThreadFactory virtualThreadFactory() {
```

Comment:
This requires following patch to Helidon (4.0.9 or similar):
```diff
git diff
diff --git webserver/webserver/src/main/java/io/helidon/webserver/ServerListener.java webserver/webserver/src/main/java/io/helidon/webserver/ServerListener.java
index c5e0761f7a..b2213ae649 100644
--- webserver/webserver/src/main/java/io/helidon/webserver/ServerListener.java
+++ webserver/webserver/src/main/java/io/helidon/webserver/ServerListener.java
@@ -60,6 +60,7 @@ import static java.lang.System.Logger.Level.DEBUG;
 import static java.lang.System.Logger.Level.ERROR;
 import static java.lang.System.Logger.Level.INFO;
 import static java.lang.System.Logger.Level.TRACE;
+import java.util.concurrent.ThreadFactory;
 
 class ServerListener implements ListenerContext {
     private static final System.Logger LOGGER = System.getLogger(ServerListener.class.getName());
@@ -143,12 +144,10 @@ class ServerListener implements ListenerContext {
                 .unstarted(this::listen);
 
         // to read requests and execute tasks
-        this.readerExecutor = ThreadPerTaskExecutor.create(Thread.ofVirtual()
-                                                                   .factory());
+        this.readerExecutor = ThreadPerTaskExecutor.create(virtualThreadFactory());
 
         // to do anything else (writers etc.)
-        this.sharedExecutor = Executors.newThreadPerTaskExecutor(Thread.ofVirtual()
-                                                                         .factory());
+        this.sharedExecutor = Executors.newThreadPerTaskExecutor(virtualThreadFactory());
 
         this.closeFuture = new CompletableFuture<>();
 
@@ -166,6 +165,10 @@ class ServerListener implements ListenerContext {
         ith.start();
     }
 
+    private static ThreadFactory virtualThreadFactory() {
+        return Thread.ofVirtual().factory();
+    }
+
     @Override
     public MediaContext mediaContext() {
         return mediaContext;
```
with such a change `ydoc-server/buildNativeImage` succeeds and starts.

Reply:
Proposal for official changes in helidon reported as https://github.com/helidon-io/helidon/pull/9271

Reply:
The [proposal was accepted](https://github.com/helidon-io/helidon/pull/9271), so the code is now in 7f7738b

Reply:
Updated to Helidon 4.1.2 in ee82801